src/dialogs/analytics/dialog-analytics-optin.ts

```diff
@@ -0,0 +1,165 @@
+import "../../components/ha-analytics";
+import "@material/mwc-button/mwc-button";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
+import { Analytics, setAnalyticsPreferences } from "../../data/analytics";
+import { haStyleDialog } from "../../resources/styles";
+import type { HomeAssistant } from "../../types";
+import { DialogAnalyticsOptInParams } from "./show-dialog-analytics-optin";
+import { analyticsLearnMore } from "../../components/ha-analytics-learn-more";
+
+@customElement("dialog-analytics-optin")
+class DialogAnalyticsOptIn extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _error?: string;
+
+  @internalProperty() private _submitting = false;
+
+  @internalProperty() private _showPreferences = false;
+
+  @internalProperty() private _analyticsDetails?: Analytics;
+
+  public async showDialog(params: DialogAnalyticsOptInParams): Promise<void> {
+    this._error = undefined;
+    this._submitting = false;
+    this._analyticsDetails = params.analytics;
+  }
+
+  public closeDialog(): void {
+    this._error = undefined;
+    this._submitting = false;
+    this._showPreferences = false;
+    this._analyticsDetails = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._analyticsDetails) {
+      return html``;
+    }
+    return html`
+      <ha-dialog
+        open
+        heading="Analytics"
+        scrimClickAction
+        escapeKeyAction
+        hideActions
+      >
+        <div class="content">
+          ${this._error ? html` <div class="error">${this._error}</div> ` : ""}
+          ${this._showPreferences
+            ? html`<ha-analytics
+                @analytics-preferences-changed=${this._preferencesChanged}
+                .hass=${this.hass}
+                .analytics=${this._analyticsDetails!}
+              ></ha-analytics>`
+            : html` <div class="introduction">
+                To help us better understand how you use Home Assistant, and to
+                ensure our priorities align with yours. We ask that you share
+                anonymized information from your installation to help make Home
+                Assistant better and help us convince manufacturers to add local
+                control and privacy-focused features.
```

Comment:
```suggestion
                To help us better understand how you use Home Assistant, and to
                ensure our priorities align with yours, we ask that you share
                anonymized information from your installation. This will help make Home
                Assistant better and help us convince manufacturers to add local
                control and privacy-focused features.
```